python/lsst/ctrl/oods/msgIngester.py

```diff
@@ -140,6 +145,9 @@ def stop_tasks(self):
             task.cancel()
         self.tasks = []
 
+    def filter_by_regex(self, files):
+        return [s for s in files if self.regex.search(s)]
```

Comment:
Since you are creating `ResourcePath` eventually another approach is not to use a regex at all but to say

```python
EXTENSIONS = (".fits", ".fits.fz")
...

def filter_by_extension(self, uris: list[ResourcePath]) -> list[ResourcePath]:
    return [s for s in uris if s.getExtension() in EXTENSIONS]
```

Reply:
(your envvar could be `.fits,.fits.gz` or something)

Reply:
I based this on the embargo-butler code at K-T's suggestion:  https://github.com/lsst-dm/embargo-butler/blob/ae7c7f555ebfec98ddf4fbb2214a53d8cc1ced96/src/enqueue.py#L42 and https://github.com/lsst-dm/embargo-butler/blob/ae7c7f555ebfec98ddf4fbb2214a53d8cc1ced96/src/enqueue.py#L66.  I believe this is because we'd like the flexibility of not only relying on extensions.